### DISCUSSION TITLE
How to extract layer from xcf gimp and masks

### DISCUSSION THREAD

**Discussion by wiso** [ImageMagick/ImageMagick#6134] [Help]
### ImageMagick version

 6.9.12-77

### Operating system, version and so on

Fedora Linux

### Description

Hello, I learn that with ImageMagic I can extract single layer from a xcf file created with gimp.

```
 convert composition.xcf[2] layer2.jpeg
```

In addition I would like also to extract the mask associated to each layer, on in alternative the image of a layer with the mask already applied. How to extract the masks?

Here a small example with two layers ("Level1" and "Level2"). Level2 has a black and white mask.
[example.xcf.zip](https://github.com/ImageMagick/ImageMagick/files/10895773/example.xcf.zip)

**Comment by snibgo** [comment]
I don't think IM can read masks in XCF files. But it might help if you posted a sample XCF.

**Reply by wiso** [reply]
I have a added a small example in the original post. I had to zip it since github doesn't allow me for xcf files.

**Comment by snibgo** [comment]
The source code coders\xcf.c function ReadOneLayer() has a stub for reading a mask, but it does nothing.

What Gimp calls "layers", IM calls "images".

In Gimp, a layer can have one mask, and the mask has only one channel. If IM were able to read a mask, what should it do? I think it should put the contents of the mask into the ReadMaskChannel of the image of the layer.

A developer may comment.
